tp1/tp1.go

```diff
@@ -24,17 +54,68 @@ type Carrito struct {
 // para cada super mercado, cuanto saldria comprar esos productos ahi.
 // Retorna un slice de carritos, donde se tiene uno para cada super mercado.
 func (p Productos) CalcularPrecios(ids ...int) []Carrito {
-	return nil
+	var carrito []Carrito
+	supermarkets := make(map[string]int)
+
+	for _, supermarket_name := range Supermercados {
```

Comment:
Podria llegar a ser que `Supermercados` tenga mas super que el input que recibimos, no? Por ejemplo, en uno de los casos de tests, `p` esta vacio. En ese caso vamos a estar iterando sobre algo que en verdad no deberiamos.

Se te ocurre alguna forma en la que se puedan armar los carritos sin utilizar `Supermercados` y solo depender de nuestra estructura `p`? Si logramos eso tambien reducimos un for loop 😉